apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryRow.cs

```diff
@@ -44,11 +44,19 @@ internal BigQueryRow(TableRow rawRow, TableSchema schema)
         private static readonly Func<string, string> StringConverter = v => v;
         private static readonly Func<string, long> Int64Converter = v => long.Parse(v, CultureInfo.InvariantCulture);
         private static readonly Func<string, double> DoubleConverter = v => double.Parse(v, CultureInfo.InvariantCulture);
-        private static readonly Func<string, DateTime> DateTimeConverter = v => new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc).AddSeconds(DoubleConverter(v));
+        // AddSeconds rounds to the nearest millisecond, for some reason.
+        private static readonly Func<string, DateTime> TimestampConverter = v =>
+        {
+            double seconds = DoubleConverter(v);
+            long microseconds = (long) (seconds * 1000000);
+            long ticks = microseconds * 10;
+            return new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc).AddTicks(ticks);
+        };
```

Comment:
I don't really understand this change - it appears to be doing the same as before? Or is it more precise somehow?